config-dro/default/kustomization.yaml

```diff
@@ -0,0 +1,91 @@
+# Adds namespace to all resources.
+namespace: ramen-system
+
+# Value of this field is prepended to the
+# names of all resources, e.g. a deployment named
+# "wordpress" becomes "alices-wordpress".
+# Note that it should also match with the prefix (text before '-') of the namespace
+# field above.
+namePrefix: ramen-dro-
+
+# Labels to add to all resources and selectors.
+#commonLabels:
+#  someName: someValue
+
+bases:
+- ../crd
+- ../rbac
+- ../../config/manager
+# [WEBHOOK] To enable webhook, uncomment all the sections with [WEBHOOK] prefix including the one in
+# crd/kustomization.yaml
+#- ../webhook
+# [CERTMANAGER] To enable cert-manager, uncomment all sections with 'CERTMANAGER'. 'WEBHOOK' components are required.
+#- ../certmanager
+# [PROMETHEUS] To enable prometheus monitor, uncomment all sections with 'PROMETHEUS'.
+#- ../prometheus
+
+patchesStrategicMerge:
+# Protect the /metrics endpoint by putting it behind auth.
+# If you want your controller-manager to expose the /metrics
+# endpoint w/o any authn/z, please comment the following line.
+- ../../config/default/manager_auth_proxy_patch.yaml
+
+# Mount the controller config file for loading manager configurations
+# through a ComponentConfig type
+#- manager_config_patch.yaml
+
+# [WEBHOOK] To enable webhook, uncomment all the sections with [WEBHOOK] prefix including the one in
+# crd/kustomization.yaml
+#- manager_webhook_patch.yaml
+
+# [CERTMANAGER] To enable cert-manager, uncomment all sections with 'CERTMANAGER'.
+# Uncomment 'CERTMANAGER' sections in crd/kustomization.yaml to enable the CA injection in the admission webhooks.
+# 'CERTMANAGER' needs to be enabled to use ca injection
+#- webhookcainjection_patch.yaml
+
+patchesJson6902:
+- target:
+    group: apps
+    version: v1
+    kind: Deployment
+    name: controller-manager
```

Comment:
Yes, the name is a bit too long. While at this, should we name/rename anything else?

Reply:
Or maybe:
ramen-drmanager-controller
ramen-drorchestrator-controller

Reply:
This is not trivial and there are multiple resources named as follows,
```
$ kubectl get clusterroles -n ramen-system | grep ramen
ramen-drm-manager-role                                                 2021-07-14T19:42:39Z
ramen-drm-metrics-reader                                               2021-07-14T19:42:39Z
ramen-drm-proxy-role                                                   2021-07-14T19:42:39Z
ramen-dro-manager-role                                                 2021-07-14T19:42:37Z
ramen-dro-metrics-reader                                               2021-07-14T19:42:37Z
ramen-dro-proxy-role                                                   2021-07-14T19:42:37Z

$ kubectl get clusterrolebinding -n ramen-system | grep ramen
ramen-drm-manager-rolebinding                          ClusterRole/ramen-drm-manager-role                                                 4m27s
ramen-drm-proxy-rolebinding                            ClusterRole/ramen-drm-proxy-role                                                   4m27s
ramen-dro-manager-rolebinding                          ClusterRole/ramen-dro-manager-role                                                 4m29s
ramen-dro-proxy-rolebinding                            ClusterRole/ramen-dro-proxy-role                                                   4m29s

$ kubectl get all -n ramen-system
NAME                                                   TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
service/ramen-drm-controller-manager-metrics-service   ClusterIP   10.107.168.155   <none>        8443/TCP   6m54s
service/ramen-dro-controller-manager-metrics-service   ClusterIP   10.101.62.2      <none>        8443/TCP   6m56s

NAME                                           READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/ramen-drm-controller-manager   1/1     1            1           6m55s
deployment.apps/ramen-dro-controller-manager   1/1     1            1           6m56s
```

I really wanted to leverage the auto generated manifests, and not have to poke around there. For example this can be solved if we edit the manifests that are auto generated, but that may get overwritten each time we generate the same.

I can only rename the deployment and leave others as is (everything will get the name prefix of `ramen-drm` or the new `ramen-drmanager`). The deployments will be named as `ramen-drmanager-controller` and `ramen-drorchestrator-controller`. Is this acceptable, or should we leave it as is?

Reply:
deployment.apps/ramen-drm-controller-manager 
deployment.apps/ramen-dro-controller-manager

1. One letter difference in a long string of words is hard to process. 
2. The manager suffix can also be confusing with our dr-manager.  

Instead of dr-manager, should we use something like kluster or mcluster?
deployment.apps/ramendr-hub-manager
deployment.apps/ramendr-kluster-manager

Also note the hyphenation of ramendr-hub makes the "hub" more readable than ramen-drhub

Reply:
So only deployments need to be renamed? Leave the rest as is? Renaming everything would be laborious.

> deployment.apps/ramen-drm-controller-manager
> deployment.apps/ramen-dro-controller-manager
> 
>     1. One letter difference in a long string of words is hard to process.

Ack!

> 
>     2. The manager suffix can also be confusing with our dr-manager.

Absolutely

> 
> 
> Instead of dr-manager, should we use something like kluster or mcluster?
> deployment.apps/ramendr-hub-manager
> deployment.apps/ramendr-kluster-manager
> 
> Also note the hyphenation of ramendr-hub makes the "hub" more readable than ramen-drhub

Here is what I think (basically we are revisiting drmanager and drorchestrator notation, which is good):
**NOTE:** For ease of using auto generated semantics I prefer to retain the controller-manager suffix
ramen-hub-controller-manager
ramen-spoke-controller-manager

Reply:
Got it.  I desperately tried to avoid suggesting `spoke` in my earlier comment because I think it is a misuse of the word spoke to refer to a managed cluster. Spoke is used to describe the path or edge between the hub and the nodes. But the alternative of `spoke nipple` (bicycle term) doesn't help either!  If you can't think of an alternative, spoke is fine with me; it matches with ACM's use of the term.

Reply:
@BenamarMk or others, one last chance (for now) before I adapt to the hub/spoke names.... 😄

Reply:
so hub and spoke I think sound good. But I think they are too heavy for ramen functionality.  
If we go with drm and dro, the drm will confuse people with digital right management that has been used in the past heavily. It is less used now adays though.  so I really don't want to go back to square 1, but I still think that if we drop manager from the end would be better.  And instead, it would be:
```
ramen-dro-controller
ramen-drmc-controller
```